lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -181,6 +198,131 @@ def test_forbidden_user_access_coach_dashboard(self):
 
 
 @attr('shard_1')
+@override_settings(
+    XBLOCK_FIELD_DATA_WRAPPERS=['lms.djangoapps.courseware.field_overrides:OverrideModulestoreFieldData.wrap'],
+    MODULESTORE_FIELD_OVERRIDE_PROVIDERS=['ccx.overrides.CustomCoursesForEdxOverrideProvider'],
+)
+@patch('xmodule.x_module.XModuleMixin.get_children', patched_get_children, spec=True)
+class TestCCXProgressChanges(CcxTestCase, LoginEnrollmentTestCase):
+    """
+    Tests ccx schedule changes in progress page
+    """
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(TestCCXProgressChanges, self).setUp()
+        start = datetime.datetime(2016, 7, 1, 0, 0, tzinfo=tzutc())
+        due = datetime.datetime(2016, 7, 8, 0, 0, tzinfo=tzutc())
+
+        self.course = course = CourseFactory.create(enable_ccx=True, start=start)
+        chapter = ItemFactory.create(start=start, parent=course, category=u'chapter')
+        sequential = ItemFactory.create(
+            parent=chapter,
+            start=start,
+            due=due,
+            category=u'sequential',
+            metadata={'graded': True, 'format': 'Homework'}
+        )
+        vertical = ItemFactory.create(
+            parent=sequential,
+            start=start,
+            due=due,
+            category=u'vertical',
+            metadata={'graded': True, 'format': 'Homework'}
+        )
+
+        # Trying to wrap the whole thing in a bulk operation fails because it
+        # doesn't find the parents. But we can at least wrap this part...
+        with self.store.bulk_operations(course.id, emit_signals=False):
+            flatten([ItemFactory.create(
+                parent=vertical,
+                start=start,
+                due=due,
+                category="problem",
+                data=StringResponseXMLFactory().build_xml(answer='foo'),
+                metadata={'rerandomize': 'always'}
+            )] for _ in xrange(2))
+
+    @patch('ccx.views.render_to_response', intercept_renderer)
+    @patch('courseware.views.views.render_to_response', intercept_renderer)
+    @patch.dict('django.conf.settings.FEATURES', {'CUSTOM_COURSES_EDX': True})
+    def test_edit_schedule(self):
+        """
+        Get CCX schedule, modify it, save it.
+        """
+        self.make_coach()
+        ccx = self.make_ccx()
+        ccx_course_key = CCXLocator.from_course_locator(self.course.id, unicode(ccx.id))
+        self.client.login(username=self.coach.username, password="test")
+
+        url = reverse('ccx_coach_dashboard', kwargs={'course_id': ccx_course_key})
+        response = self.client.get(url)
+
+        schedule = json.loads(response.mako_context['schedule'])  # pylint: disable=no-member
+        self.assertEqual(len(schedule), 1)
+
+        def unhide(unit):
```

Comment:
please move this right after the definition of the test (at least)

though I would make it a class method even if it is used by only this test.